modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ListTypeDefinition/ListTypeEntriesModal.tsx

```diff
@@ -82,7 +82,7 @@ function ListTypeEntriesModal() {
 			...previousValues,
 			...(modalType !== 'edit' &&
 				keyChanged === false &&
-				name_i18n?.[defaultLanguageId] && {
+				typeof name_i18n?.[defaultLanguageId] === 'string' && {
```

Comment:
Hey @devpedrotavares, I think we can remove this `name_i18n?.[defaultLanguageId` validation since this value always comes as a `string`, we can also make this code more readable by removing all this validation and putting it on the line above like:


```ts
const handleNameChange = (newName_i18n: LocalizedValue<string>) => {
    let newItemKey = itemKey;
    
    if (modalType !== 'edit' && keyChanged === false) {
	    newItemKey = toCamelCase(newName_i18n[defaultLanguageId] as string);
    }
    
    setState((previousValues) => ({
	    ...previousValues,
	    itemKey: newItemKey,
	    name_i18n: newName_i18n,
    }));
};

```

Also, where this function is passed, we can remove the function execution and pass only the function reference:

```tsx
onChange={handleNameChange}
```

Reply:
Hi @danvb112, I did as you instructed. It looks better!